knimin/tests/test_ag_edit_barcode.py

```diff
@@ -0,0 +1,108 @@
+from unittest import main
+
+from tornado.escape import url_escape
+
+from knimin.tests.tornado_test_base import TestHandlerBase
+from knimin import db
+
+
+class TestAGEditBarcodeHandler(TestHandlerBase):
+    def test_get_not_authed(self):
+        response = self.get('/ag_edit_barcode/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/login/?next=%s' %
+                         (port, url_escape('/ag_edit_barcode/')))
+
+    def test_get(self):
+        self.mock_login_admin()
+
+        # check that error is raised for unknown barcode
+        response = self.get('/ag_edit_barcode/', {'barcode': 'unknown'})
+        self.assertEqual(response.code, 500)
+        self.assertRaises(KeyError)
```

Comment:
surprisingly it works :-) Could it be that first a page is returned with code 500 and after that an exception is raised by tornado? Anyways, I don't see a reason to change the code here, do you agree?